geowave-store/src/main/java/mil/nga/giat/geowave/store/filter/SpatialQueryFilter.java

```diff
@@ -188,7 +193,7 @@ public void fromBinary(
 					fieldId));
 		}
 		buf.get(theRest);
-		queryGeometry = GeometryUtils.geometryFromBinary(geometryBinary);
+		queryGeometry = geometryInterner.intern(GeometryUtils.geometryFromBinary(geometryBinary));
```

Comment:
```Java
@Test
	public void testInterning()
			throws ParseException {
		final Geometry g = GeometryUtils.GEOMETRY_FACTORY.createPolygon(new Coordinate[] {new Coordinate(0,0), new Coordinate(1,0), new Coordinate(1,1), new Coordinate(0,1), new Coordinate(0,0)});
		final Geometry gNewInstance = GeometryUtils.GEOMETRY_FACTORY.createPolygon(new Coordinate[] {new Coordinate(0,0), new Coordinate(1,0), new Coordinate(1,1), new Coordinate(0,1), new Coordinate(0,0)});
		final WKBWriter wkbWriter = new WKBWriter();
		final byte[] b = wkbWriter.write(g);
		final byte[] b2 = new byte[b.length];
		System.arraycopy(b, 0, b2, 0, b.length);
		final WKBReader wkbReader = new WKBReader();
		final Geometry gSerialized = wkbReader.read(b);
		final Geometry gSerlizedArrayCopy = wkbReader.read(b2);


		System.out.println("Orig = new instance, same constructor: " + g.equals(gNewInstance));
		System.out.println("Orig  -> wkb -> geom equals: " + g.equals(gSerialized));
		System.out.println("Orig -> wkb -> arraycopy -> geom equals: " + g.equals(gSerlizedArrayCopy));
		System.out.println("(Orig -> wkb -> geom) = (wkb -> arraycopy -> geom): " + gSerialized.equals(gSerlizedArrayCopy));
	}
```

prints:
```
Orig = new instance, same constructor: true
Orig  -> wkb -> geom equals: true
Orig -> wkb -> arraycopy -> geom equals: true
(Orig -> wkb -> geom) = (wkb -> arraycopy -> geom): true
```

which seems to disagree with comment above, but explain why the interner worked; digging into it.